### DISCUSSION TITLE
Long running query times out

### DISCUSSION THREAD

**Discussion by zrd740** [burningalchemist/sql_exporter#204] [Q&A] [answered]
**Describe the bug**
I have a MSSQL query that takes about 2 minutes to complete. One of my main reasons for wanting to use sql-exporter is so that I can pull the results of all these queries into Prometheus so I can see the values in grafana without having to wait for the queries to run every time I want to refresh a dashboard, also so I can see how the values are changing over time.

When I try and run the query through sql_exporter, I get:
ts=2023-02-17T00:25:21.925Z caller=klog.go:84 level=debug func=Infof msg="Error gathering metrics: [from Gatherer #1] [collector=\"qb-test\", query=\"ar_balance\"] context deadline exceeded"

I can see that the query is running, and it gets this error before the query is expected to return the data.

**To Reproduce**
Try to run a query longer than 20 seconds, and you will get this error.

**Expected behavior**
I couldn't find a way to set the timeout for running a query, it would be great to have this in the .yml file so that it could be adjusted for more use cases.

**Comment by burningalchemist** [comment, marked as answer]
Hey @zrd740,

In the documentation folder you can find the config file with some descriptions, please have a look.

As for `scrape_timeout` please try the following:
```
# Global settings and defaults.
global:
  # Scrape timeouts ensure that:
  #   (i)  scraping completes in reasonable time and
  #   (ii) slow queries are canceled early when the database is already under heavy load
  # Prometheus informs targets of its own scrape timeout (via the "X-Prometheus-Scrape-Timeout-Seconds" request header)
  # so the actual timeout is computed as:
  #   min(scrape_timeout, X-Prometheus-Scrape-Timeout-Seconds - scrape_timeout_offset)
  #
  # If scrape_timeout <= 0, no timeout is set unless Prometheus provides one. The default is 10s.
  scrape_timeout: 10s
```

Also check the scrape_timeout on the Prometheus side.

**Comment by burningalchemist** [comment]
I'll convert this issue into a discussion since there's no response. Currently, not a bug.
